content/articles/building-an-ensemble-learning-based-regression-model-using-python/index.md

```diff
@@ -0,0 +1,173 @@
+
+For our model's evaluation, we shall be using RMSE (Root Mean Squared Error). 
+
+> **NB:** Regression problems cannot be measured using accuracy metric since the goal is to measure how close the predicted values is to the expected values and not to evaluate how correct the prediction is. Hence we use errors to evaluate our models.
+
+### Prerequisites
+
+To follow through the tutorial, you need to:
+1. Know the basics of Python
+2. Have a [Kaggle](https://www.kaggle.com/) account
+3. Know basics of Machine Learning
+
+### Setting Up Your Environment
+
+Before we start building our model, we shall first go to Kaggle and create a new **notebook** and rename it to **Create_Folds**.
+![notebook](engineering-education/building-an-ensemble-learning-based-regression-model-using-python/notebook.jpg)
+ 
+After that, [download](https://www.kaggle.com/c/30-days-of-ml/data) the data from Kaggle and add it to your environment using the **Add Data** button and upload the downloaded data as **Dataset**.
+
+>**HINT:**  To flawlessly upload your data to Kaggle, compress the datasets.
+
+### Creating K-Folds
+Once done with setting up the environment, we shall move on to creating k-folds for our dataset.
+Cross-validation is a validation technique used to evaluate machine learning models on a finite dataset.  It is quite popular as it is easier to understand and results to less biased predictions than other methods like train/test split.
+
+It is also best that whenever you're starting with a machine learning problem, you create folds that you'll be using throughout the modeling process.
+
+#### Importing the necessary Libraries
+
+Before proceeding, we need to import the following necessary libraries.
+
+```python
+import numpy as np 
+import pandas as pd 
+from sklearn import model_selection
+```
+#### Read Data
+```python
+train_data = pd.read_csv('/kaggle/input/Dataset/train.csv')
+test_data = pd.read_csv('/kaggle/input/Dataset/test.csv')
+submission = pd.read_csv('/kaggle/input/Dataset/sample_submission.csv')
+```
+#### Creating Folds
+Let's create a new column with the name *kfold* on the last column as shown below.
+```python
+train_data['kfold'] = -1
+```
+We shall then proceed to create 5 folds using the following code block:
+
+```python
+kf  = model_selection.KFold(n_splits= 5,shuffle = True, random_state=42)
+
+for fold, (train_indicies,valid_indicies)in enumerate(kf.split(X=train_data)):
+    train_data.loc[valid_indicies, "kfold"]=fold
+```
+
+After running the cell above, we shall output the *new* csv file (`train_kfolds.csv`) with kfolds by running the code block below:
+```python
+train_data.to_csv('train_kfolds.csv', index=False)
+```
+### Building a Regression Model
+
+After creating the kfolds, we shall proceed to download the `train_kfolds.csv` from the output data on our **Create_kFolds** notebook. 
+
```

Comment:
Make sure to replace all the instances you've used `shall` in your tutorial with `will`